modules/pam_exec/pam_exec.c

```diff
@@ -180,12 +180,12 @@ call_exec (const char *pam_type, pam_handle_t *pamh,
 	      if (resp)
 		{
 		  pam_set_item (pamh, PAM_AUTHTOK, resp);
-		  authtok = strndupa (resp, PAM_MAX_RESP_SIZE);
+		  strcpy (authtok, resp);
 		  _pam_drop (resp);
 		}
 	    }
 	  else
-	    authtok = strndupa (void_pass, PAM_MAX_RESP_SIZE);
+	    strcpy (authtok, void_pass);
```

Comment:
This is unsafe. Although the password if obtained from conversation function cannot be larger than PAM_MAX_RESP_SIZE, in theory some module can set the PAM_AUTHTOK item to any arbitrarily large string. So to be on the safe side, please use strncpy correctly and do it in the above call as well.